src/packages/name-service/blockstack-service.ts

```diff
@@ -170,8 +182,9 @@ export class BlockstackService extends nameService.NameService {
         await this._gaiaService.uploadProfileInfo(identityKeyPair.privKey);
 
         const registeredSubdomain = await this._registerSubdomain(subdomain, identityKeyPair.address);
+        const cruxdomain = this._domain.slice(0, -5);
```

Comment:
This should be the responsibility of identityUtils package. You can externalise this as a method in identityUtils package and use the corresponding method here.